### PR TITLE
feat: kubernetes relationship by lookup

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/config-db/issues/320

### CODE REVIEW

File: db/update.go [hunk 1]
```diff
@@ -359,14 +361,19 @@ func relationshipResultHandler(relationships v1.RelationshipResults) error {
...
+		if relationship.RelatedConfigID != uuid.Nil {
+			relatedID = commonUtils.Ptr(relationship.RelatedConfigID.String())
+		} else {
+			relatedID, err = FindConfigItemID(relationship.RelatedExternalID)
+			if err != nil {
+				logger.Errorf("error fetching external config item(id=%s): %v", relationship.RelatedExternalID, err)
+				continue
+			}
+			if relatedID == nil {
+				logger.Warnf("related external config item(id=%s) not found.", relationship.RelatedExternalID)
```
Comment (moshloop) [inline db/update.go:374]:
```suggestion
				logger.Debugf("related external config item(id=%s) not found.", relationship.RelatedExternalID)
```

**Comment by moshloop** [comment]
@adityathebe  Could we update the example fixture and maybe add an integration test?

File: db/config.go [hunk 2]
```diff
@@ -87,6 +89,19 @@ func UpdateConfigItem(ci *models.ConfigItem) error {
 	return nil
 }
 
+// ConfigItemsIDs returns the uuid of config items which matches the given type, name & namespace
+func ConfigItemsIDs(ctx context.Context, configType, name, namespace string) ([]uuid.UUID, error) {
```
Comment (moshloop) [inline db/config.go:93]: Shouldn't this be FindConfigByNamespaceName?

File: scrapers/runscrapers_test.go [hunk 3]
```diff
@@ -26,14 +30,85 @@ var _ = Describe("Scrapers test", func() {
...
+			scraperCtx := api.NewScrapeContext(gocontext.TODO(), gormDB, nil).WithScrapeConfig(&scrapeConfig)
+			_, err := RunScraper(scraperCtx)
+			Expect(err).To(BeNil())
+
+			var configRelationships []models.ConfigRelationship
+			err = gormDB.Find(&configRelationships).Error
+			Expect(err).To(BeNil())
+
+			Expect(len(configRelationships)).To(Equal(1))
+			Expect(configRelationships[0].Relation).To(Equal("ConfigMapConfigMap"))
```
Comment (moshloop) [inline scrapers/runscrapers_test.go:108]: I think this should just be ConfigMap

Comment (adityathebe) [reply scrapers/runscrapers_test.go:108]: Aren't we concatenating the kind of the two related objects? Eg (from azure demo):
```       
PodConfigMap                    
NodePod           
GrafanaConfigMap         
EKSNode                         
DaemonSetPod
```

Comment (moshloop) [reply scrapers/runscrapers_test.go:108]: We shouldn't be, but we can tackle that later

Comment (adityathebe) [reply scrapers/runscrapers_test.go:108]: https://github.com/flanksource/config-db/issues/323

**Pull request merged**